src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs

```diff
@@ -127,37 +127,45 @@ public async Task<IActionResult> LoginPOST(string returnUrl = null)
             if (user != null)
             {
                 var result = await _signInManager.CheckPasswordSignInAsync(user, model.Password, lockoutOnFailure: true);
+
                 if (result.Succeeded)
                 {
-                    foreach (var handler in _accountEvents)
+                    foreach (var accountEvent in _accountEvents)
                     {
-                        var loginResult = await handler.LoggingInAsync(user);
+                        var loginResult = await accountEvent.ValidatingLoginAsync(user);
 
                         if (loginResult != null)
                         {
                             return loginResult;
                         }
                     }
 
-                    result = await _signInManager.PasswordSignInAsync(user, model.Password, model.RememberMe, lockoutOnFailure: true);
+                    await _accountEvents.InvokeAsync((e, user, modelState) => e.LoggingInAsync(user.UserName, (key, message) => modelState.AddModelError(key, message)), user, ModelState, _logger);
```

Comment:
This is now run twice, also up on ln 119. Is this deliberate?

Reply:
It's not. I'll fix that tomorrow.